control/autonomous_emergency_braking/include/autonomous_emergency_braking/node.hpp

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTONOMOUS_EMERGENCY_BRAKING__NODE_HPP_
+#define AUTONOMOUS_EMERGENCY_BRAKING__NODE_HPP_
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <pcl_ros/transforms.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <tier4_autoware_utils/tier4_autoware_utils.hpp>
+#include <vehicle_info_util/vehicle_info_util.hpp>
+
+#include <autoware_auto_planning_msgs/msg/trajectory.hpp>
+#include <autoware_auto_system_msgs/msg/autoware_state.hpp>
+#include <autoware_auto_vehicle_msgs/msg/velocity_report.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <visualization_msgs/msg/marker.hpp>
+
+#include <boost/optional.hpp>
+
+#include <pcl/common/transforms.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace autoware::motion::control::autonomous_emergency_braking
+{
+
+using autoware_auto_planning_msgs::msg::Trajectory;
+using autoware_auto_system_msgs::msg::AutowareState;
+using autoware_auto_vehicle_msgs::msg::VelocityReport;
+using nav_msgs::msg::Odometry;
+using sensor_msgs::msg::Imu;
+using sensor_msgs::msg::PointCloud2;
+using PointCloud = pcl::PointCloud<pcl::PointXYZ>;
+using diagnostic_updater::DiagnosticStatusWrapper;
+using diagnostic_updater::Updater;
+using tier4_autoware_utils::Point2d;
+using tier4_autoware_utils::Polygon2d;
+using vehicle_info_util::VehicleInfo;
+using visualization_msgs::msg::Marker;
+using visualization_msgs::msg::MarkerArray;
+using Path = std::vector<geometry_msgs::msg::Pose>;
+
+struct ObjectData
+{
+  rclcpp::Time stamp;
+  geometry_msgs::msg::Point position;
+  double velocity{0.0};
+  double rss{0.0};
+  double distance_to_object{0.0};
+};
+
+class CollisionDataKeeper
+{
+public:
+  explicit CollisionDataKeeper(rclcpp::Clock::SharedPtr clock) { clock_ = clock; }
+
+  void setTimeout(const double timeout_sec) { timeout_sec_ = timeout_sec; }
+
+  bool checkExpired()
+  {
+    if (data_ && (clock_->now() - data_->stamp).seconds() > timeout_sec_) {
+      data_.reset();
+    }
+    return (data_ == nullptr);
+  }
```

Comment:
Changing internal variables with the bool type check function is not recommended, as it may lead to misunderstandings by maintenance personnel or human error.
Since this is a backport PR, no fix action is required in this PR.